client/ayon_core/pipeline/traits/meta.py

```diff
@@ -0,0 +1,113 @@
+"""Metadata traits."""
+from typing import ClassVar, List
+
+from pydantic import Field
+
+from .trait import TraitBase
+
+
+class Tagged(TraitBase):
+    """Tagged trait model.
+
+    This trait can hold list of tags.
+
+    Example::
+
+        Tagged(tags=["tag1", "tag2"])
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+        tags (List[str]): Tags.
+    """
+
+    name: ClassVar[str] = "Tagged"
+    description: ClassVar[str] = "Tagged Trait Model"
+    id: ClassVar[str] = "ayon.meta.Tagged.v1"
+    tags: List[str] = Field(..., title="Tags")
+
+
+class TemplatePath(TraitBase):
+    """TemplatePath trait model.
+
+    This model represents a template path with formatting data.
+    Template path can be Anatomy template and data is used to format it.
+
+    Example::
+
+        TemplatePath(template="path/{key}/file", data={"key": "to"})
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+        template (str): Template path.
+        data (dict[str]): Formatting data.
+    """
+
+    name: ClassVar[str] = "TemplatePath"
+    description: ClassVar[str] = "Template Path Trait Model"
+    id: ClassVar[str] = "ayon.meta.TemplatePath.v1"
+    template: str = Field(..., title="Template Path")
+    data: dict = Field(..., title="Formatting Data")
+
+
+class Variant(TraitBase):
+    """Variant trait model.
+
+    This model represents a variant of the representation.
+
+    Example::
+
+        Variant(variant="high")
+        Variant(variant="prores444)
+
+    Attributes:
+        name (str): Trait name.
+        description (str): Trait description.
+        id (str): id should be namespaced trait name with version
+        variant (str): Variant name.
+    """
+
+    name: ClassVar[str] = "Variant"
+    description: ClassVar[str] = "Variant Trait Model"
+    id: ClassVar[str] = "ayon.meta.Variant.v1"
+    variant: str = Field(..., title="Variant")
+
+
+class KeepOriginalLocation(TraitBase):
+    """Keep files in its original location.
+
+    Note:
+        This is not a persistent trait.
+
+    """
+    name: ClassVar[str] = "KeepOriginalLocation"
+    description: ClassVar[str] = "Keep Original Location Trait Model"
+    id: ClassVar[str] = "ayon.meta.KeepOriginalLocation.v1"
+    persistent: bool = Field(default=False, title="Persistent")
+
+class KeepOriginalName(TraitBase):
+    """Keep files in its original name.
+
+    Note:
+        This is not a persistent trait.
+
+    """
+    name: ClassVar[str] = "KeepOriginalName"
+    description: ClassVar[str] = "Keep Original Name Trait Model"
+    id: ClassVar[str] = "ayon.meta.KeepOriginalName.v1"
+    persistent: bool = Field(default=False, title="Persistent")
+
+
+class SourceApplication(TraitBase):
```

Comment:
This trait is weird, why is there platform (I would expect that on some "base" trait)? Where we gonna get those information? Attributes `application` and `variant` is `ayon-applications` addon specific information, and `version` can be collected only by host integration.

Reply:
Think of it as metadata - it doesn't have any logic - only holds the information what was the application used to create this representation. This information needs to be filled by the host producing this representation.

Reply:
Well but it is trait that would combine information from 3 different sources... What if you're missing one of them?